src/backend/storage/ipc/dsm.c

```diff
@@ -14,6 +14,12 @@
  * hard postmaster crash, remaining segments will be removed, if they
  * still exist, at the next postmaster startup.
  *
+ * When invoking the low level functions, the caller must correctly
+ * set "request_size" when creating and destroying segments.
```

Comment:
Really not sure if this is going to end up flying with the community.  Also- I'm not 100% sure that we're always going to know the size before destroying, if it isn't the same process creating as destroying?  If we do know, then we must be keeping track of that somewhere and, if we're doing that, why can't we just pull that info?

Reply:
Currently, we don't seem to be keeping the info around. We do keep "mapped size" for each mapped segment, but we have a problem when deleting a segment we haven't mapped (eg. unpin). Part of this change is to stash creation size in the dsm table where we can easily access it.